src/ompl/geometric/planners/rrt/src/RRTstar.cpp

```diff
@@ -573,73 +661,210 @@ void ompl::geometric::RRTstar::getPlannerData(base::PlannerData &data) const
+        {
+            // Mark as prunable
+            pruneThisLeaf = true;
+        }
+        // No else, it does not meet the pruning condition.
     }
-    return 0;
+    // No else, it has children
+
+    return pruneThisLeaf;
+}
+
+ompl::base::Cost ompl::geometric::RRTstar::solutionHeuristic(const Motion *motion, const bool estimate) const
+{
+    base::Cost costToCome;
+    if (estimate)
+        costToCome = opt_->motionCost(startMotion_->state, motion->state); // lower-bounding cost from the start to the state
+    else
+        costToCome = motion->cost; // current cost from the state to the goal
+
+    const base::Cost costToGo = opt_->costToGo(motion->state, pdef_->getGoal().get()); // lower-bounding cost from the state to the goal
+    return opt_->combineCosts(costToCome, costToGo); // add the two costs
 }
 
-void ompl::geometric::RRTstar::deleteBranch(Motion *motion)
+void ompl::geometric::RRTstar::setSampleRejection(const bool reject)
 {
-    removeFromParent(motion);
+    // Check if we're changing the setting of rejection sampling. If we are, we will need to create a new sampler, which we only want to do if one is already allocated.
+    if (reject != useRejectionSampling_)
+    {
+        // Store the setting
+        useRejectionSampling_ = reject;
 
-    std::vector<Motion *>& toDelete = pruneScratchSpace_.candidates;
-    toDelete.clear();
-    toDelete.push_back(motion);
+        // If we currently have a sampler, we need to make a new one
+        if (sampler_ || infSampler_)
+        {
+            // Reset the samplers
+            sampler_.reset();
+            infSampler_.reset();
 
-    while (!toDelete.empty())
+            // Create the sampler
+            allocSampler();
+        }
+    }
+}
+
+void ompl::geometric::RRTstar::setInformedRrtStar(bool informedRrtStar)
+{
+    // Toggle tree pruning:
+    useTreePruning_ = informedRrtStar;
+
+    // Check if we're changing the setting of informed sampling. If we are, we will need to create a new sampler, which we only want to do if one is already allocated.
+    if (informedRrtStar != useInformedSampling_)
     {
-        Motion *mto_delete = toDelete.back();
-        toDelete.pop_back();
+        // Store the value
+        useInformedSampling_ = informedRrtStar;
 
-        for(std::size_t i = 0; i < mto_delete->children.size(); ++i)
-            toDelete.push_back(mto_delete->children[i]);
+        // If we currently have a sampler, we need to make a new one
+        if (sampler_ || infSampler_)
+        {
+            // Reset the samplers
+            sampler_.reset();
+            infSampler_.reset();
 
-        si_->freeState(mto_delete->state);
-        delete mto_delete;
+            // Create the sampler
+            allocSampler();
+        }
     }
 }
 
-ompl::base::Cost ompl::geometric::RRTstar::costToGo(const Motion *motion, const bool shortest) const
+void ompl::geometric::RRTstar::allocSampler()
```

Comment:
Create the requested sampler, either informed (which can be direct), rejection, or regular.
Since we're doing rejection like this anyway, I think we could do just informed or regular, but as I said, I wasn't ready to go that far.